### DISCUSSION TITLE
Search and Highlight Specific Categories

### DISCUSSION THREAD

**Discussion by mzeiher** [organicmaps/organicmaps#789] [Ideas]
Hi I use this feature to search and highlight specific categories a lot in Osmand, it would be nice to search for a specific osm category and highlight them on the map. 

We use this feature a lot when we travel with our van to find near campsites, sanitary disposal facilities or parking and picnic places

![Screenshot_20210702-111945_OsmAnd~](https://user-images.githubusercontent.com/1873877/124252977-d5aca100-db27-11eb-8ace-7ca7670c4e3c.png)
osmand lets you display all places with the specific category on the map
![Screenshot_20210702-111925_Organic_Maps](https://user-images.githubusercontent.com/1873877/124252980-d6ddce00-db27-11eb-990f-1e6c691ef477.png)
in organic maps it just shows you places with camp site in the name and categories

**Comment by biodranik** [comment]
Many categories are already searchable and highlighted. Please list specific ones which are missing.

**Reply by mzeiher** [reply]
Hi, good questions, it depends, if I'm planing a hike I like to see huts (tourism=alpine_hut) and shelters (amenity=shelter) or if we drive around with our van parking (amenity=parking, which is possible at the moment) or caravan sites (tourism=caravan_site) or camping sites (tourism=camp_site) it helps to be able to highlight these on the maps so that are they are visible in lower zoom steps to get a nice overview of the available POIs in a specific region.

It would be nice to make it possible to only search for a specific category (tag) and highlight them on a map instead of mixing categories and search results in the search window

**Reply by GLLM** [reply]
Earlier today, in fact, I tried to display on hte map only the category "_spot-scuba_diving_", to no avail !
I guess I had the same requirement as you have !

**Reply by mzeiher** [reply]
@gllm jep i think its the same requirement

![Screenshot_20210713-105730_OsmAnd~.png](https://user-images.githubusercontent.com/1873877/125423213-942b2db9-7175-4a25-9ea5-3b63571f7162.png)

![Screenshot_20210713-105722_OsmAnd~.png](https://user-images.githubusercontent.com/1873877/125423263-a67cbf1f-d532-4ccd-a011-3197f97d5b52.png)

**Comment by pastk** [comment]
Camp sites are searchable in OM via a "campsite", "campground" or "camping" query. We should add a "camp site" alias too.

Dump stations: "Dump Station", "sanitary dump station" and a few other terms.

You can check search aliases for other categories in https://github.com/organicmaps/organicmaps/blob/master/data/categories.txt
Also feel free to add more aliases there and send a PR!

We don't support scuba spots yet, please create an issue to add it.

**Reply by Esmolante** [reply]
I think an important category for quick searches would be to unify tourism=picnic_site and highway=rest_area. To easily find a place to stop and rest or have a snack when driving.
A hut or scuba diving are very specific, but a picnic site or a rest area is important for most drivers.

**Reply by j13m126** [reply]
@GLLM Little update: scuba diving is searchable now (version 2022-08-23) :)

**Reply by GLLM** [reply]
Thanks for the notice, appreciated.
I tested it and ... yeah : many blue points now appear ! :-)

**Comment by Esmolante** [comment]
As more highlight categories are added, the list will grow. To make it manageable, they could be grouped together. For example, a main "health" category could encompass the existing "hospital" and " pharmacy" categories. Or group bank and ATM into a "money" category. In that way, it would be possible to put the "scuba diving" category under "sport". Or huts and shelters with hotels, into category "sleep".

**Reply by Misalf-git** [reply]
I like things to be organized as well, but I believe categorizing can be difficult and may lead to confusion. Meaning it won't help if one thinks to be looking at the correct category, if whoever compiled it thought differently.

**Comment by biodranik** [comment]
OM can search for _any_ category that is visible on the map, not only for those categories in the search list:
https://github.com/organicmaps/organicmaps/blob/master/data/categories.txt

Let's focus on the important task: **how to teach users that they can type any category in the search and find it**?

**Reply by pcgaustad** [reply]
> OM can search for any category that is visible on the map, not only for those categories in the search list[.]

That does not seem to be the case. For example, the Alter Wall street in Hamburg is visible in the app as a "Pedestrian Street" (see screenshot below), but I'm not able to search for this category (i tried "pedestrian street", "pedestrian", "highway=pedestrian", "highway.pedestrian", "type.highway.pedestrian"). Or am I doing something wrong?

Is the search functionality documented somewhere?

![IMG_7320](https://github.com/user-attachments/assets/0c669796-e786-4bbb-97f5-67d3def42721)

**Reply by biodranik** [reply]
You're right, only types listed in categories.txt are searchable (we manually choose only those important for users, which users may guess to type in the search box without any hints).

**Reply by pcgaustad** [reply]
> Let's focus on the important task: **how to teach users that they can type any category in the search and find it**?

How about an option "Show all" at the bottom of the shortlist of categories which would list all searchable categories (alphabetically)? That way it would be obvious for users that they may search for a lot of categories beyond the ones in the shortlist, and they would also get to see the complete list – without overwhelming other users with a lot of options (categories) they never asked for.

**Comment by j13m126** [comment]
In order to let users know they can search for any category, the search results should indicate why they are shown. This is only partly the case right now. Here are two ideas:
1. If you search for 'bicyc' the search term is written bold in the poi's name. But not in the poi's type. I think it would help if the poi type would also be written in bold or highlighted in some way. ![Screenshot_20220914-002305.png](https://user-images.githubusercontent.com/7961926/190021142-de9bf3bd-5f6d-4117-a972-c3d7a24edb63.png)

2. Categories in search suggestions should be distinguishable from other suggestions like places or street names. They could be, for instance, written inside a box. ![Screenshot_20220914-001304~2.png](https://user-images.githubusercontent.com/7961926/190021218-749bbd6a-85a5-4bd3-94d4-4ccbe1c60c8f.png)

**Reply by biodranik** [reply]
Wow, thanks! These two ideas are _very_ good! CC @vng @arnaudvergnet
